numpy/_core/include/numpy/ndarraytypes.h

```diff
@@ -162,18 +162,39 @@ enum NPY_TYPECHAR {
 };
 
 /*
- * Changing this may break Numpy API compatibility
- * due to changing offsets in PyArray_ArrFuncs, so be
- * careful. Here we have reused the mergesort slot for
- * any kind of stable sort, the actual implementation will
- * depend on the data type.
+ * Changing this may break Numpy API compatibility due to changing offsets in
+ * PyArray_ArrFuncs, so be careful. Here we have reused the mergesort slot for
+ * any kind of stable sort, the actual implementation will depend on the data
+ * type.
+ *
+ * Updated in NumPy 2.4
+ *
+ * Updated with new names denoting requirements rather than the algorithm. All
+ * the previous values are reused in a way that should be downstream
+ * compatible, but the actual algorithms used may be different than before. The
+ * new approach should be more flexible and easier to update. The idea is that
```

Comment:
No big deal but would delete the sentence "The idea is ..." - the implementation should not matter here (I may well end up thinking that this is the right approach, and now see how indexing is sort-of baked in to the present API, but not having seeing any implementation, I still feel the logical solution would be a single slot for a function with a flags argument...).